opendevin/server/listen.py

```diff
@@ -137,6 +138,22 @@ def select_file(file: str):
     return {'code': content}
 
 
+@app.post('/api/upload-file')
+async def upload_file(file: UploadFile):
+    try:
+        workspace_base = config.get(ConfigType.WORKSPACE_BASE)
+        file_path = Path(workspace_base) / file.filename
```

Comment:
We need to add some validation to `filename` here, to make sure it's not relative to the workspace base